> Instead faced with a linker error: `LINK : fatal error LNK1181: cannot open input file 'Catch2d.lib.windows.opt.tools.64.lib'`

that's weird, because it seems like a LIBSUFFIX problem, but I would expect to see the same problem on unix

--- a/SConstruct
+++ b/SConstruct
@@ -321,7 +321,7 @@ if env_base["build_custom_tests"]:
 	if sys.platform == "win32":
 		env_base.Append(CPPPATH=["C:\\Program Files (x86)\\catch2\\include"])
 		env_base.Append(LIBPATH=["C:\\Program Files (x86)\\catch2\\lib"])
-		env_base.Append(LIBS=["Catch2d.lib"])
+		env_base.Append(LIBS=[File("Catch2d.lib")])
 	else:
 		env_base.Append(CPPPATH=["/usr/local/include"])
 		env_base.Append(LIBPATH=["/usr/local/lib"])

--- a/SConstruct
+++ b/SConstruct
@@ -322,7 +322,7 @@ if env_base["build_custom_tests"]:
 		env_base.Append(CPPPATH=["C:\\Program Files (x86)\\catch2\\include"])
 		env_base.Append(LIBPATH=["C:\\Program Files (x86)\\catch2\\lib"])
 		env_base.Append(
-		    LIBS=[File("C:\\Program Files (x86)\\catch2\\lib\\Catch2d.lib")])
+		    LIBS=[File("C:\\Program Files (x86)\\catch2\\lib\\Catch2.lib")])
 	else:
 		env_base.Append(CPPPATH=["/usr/local/include"])
 		env_base.Append(LIBPATH=["/usr/local/lib"])

--- a/SConstruct
+++ b/SConstruct
@@ -321,7 +321,8 @@ if env_base["build_custom_tests"]:
 	if sys.platform == "win32":
 		env_base.Append(CPPPATH=["C:\\Program Files (x86)\\catch2\\include"])
 		env_base.Append(LIBPATH=["C:\\Program Files (x86)\\catch2\\lib"])
-		env_base.Append(LIBS=[File("Catch2d.lib")])
+		env_base.Append(
+		    LIBS=[File("C:\\Program Files (x86)\\catch2\\lib\\Catch2d.lib")])
 	else:
 		env_base.Append(CPPPATH=["/usr/local/include"])
 		env_base.Append(LIBPATH=["/usr/local/lib"])

--- a/platform/windows/godot_windows.cpp
+++ b/platform/windows/godot_windows.cpp
@@ -178,7 +178,7 @@ __declspec(dllexport) int widechar_main(int argc, wchar_t **argv) {
 		List<String> args = os.get_cmdline_args();
 		if (args.find("--test-external") != nullptr) {
 			#ifdef CATCH_TESTS
-			run_catch_tests(argc, argv);
+			run_catch_tests(argc, &argv_utf8[0]);
 			#else
 			printf("Option --test_external is invalid because this program was built without Catch2.");
 			#endif

